edk2toollib/windows/telem/cper_parser_tool.py

```diff
@@ -0,0 +1,659 @@
+            return "Fatal"
+        elif(self.sectionSeverity == 2):
+            return "Corrected"
+        elif(self.sectionSeverity == 3):
+            return "Informational"
+
+        return ""
+
+    def FruStringParse(self) -> str:
+        '''Parse out the custom string identifying the Fru hardware'''
+
+        # Only parse if data is valid
+        if(self.fruStringValid):
+
+            # Convert the Fru string from bytes to a string
+            try:
+                return "".join([chr(x) for x in self.fruString])
+            except:
+                return "Unable to parse"
+
+        return "Invalid"
+
+    def PrettyPrint(self) -> str:
+        '''Print relevant portions of the section header. Change to suit your needs'''
+
+        string = ""
+        temp = ""
+
+        string += "Section Length:      " + str(self.sectionLength) + '\n'
+
+        temp = self.FlagsParse()
+        string += "Flags:               " + temp + '\n'
+
+        temp = self.SectionSeverityParse()
+        string += "Section Severity:    " + temp + '\n'
+
+        temp = self.SectionTypeParse()
+        string += "Section Type:        " + temp + '\n'
+
+        temp = self.FruIdParse()
+        string += "Fru Id:              " + temp + '\n'
+
+        temp = self.FruStringParse()
+        string += "Fru String:          " + temp + '\n'
+
+        return string
+
+
+def HexDump(input: bytes, bytesperline: int) -> str:
+    '''Dumps byte code of input'''
+
+    string = ""  # Stores the entire hexdump string
+    asc = ""  # Stores the ascii version of the current hexdump line
+    byte = ""  # Stores the base 16 byte version of the current hexdump line
+
+    # Used to check if a byte value is within relevant ascii character bounds
+    def rangecheck(x):
+        return x < 31 or x > 127
+
+    # Go through every byte from the input
+    for i in range(len(input)):
+
+        # Add a byte string version of the byte onto the byte string
+        byte += format(input[i], '02X') + " "
+
+        # Add an ascii version of the byte onto the ascii string
+        if rangecheck(input[i]):
+            asc += ".  "
+        else:
+            asc += format(chr(input[i]), " <2") + " "
+
+        # Once we've reached bytesperline length, concatenate asc and byte strings and start a new line
+        if(not (i + 1) % bytesperline):
+            string += byte + " " + asc + "\n"
+            asc = ""
+            byte = ""
+
+    # Check if there are any remaining characters in the asc and byte strings to be added to string
+    if(len(input) % bytesperline):
+        string += byte + "   " * (bytesperline - (len(byte) // 3)) \
+            + " " + asc + "   " * (bytesperline - (len(asc) // 3)) + "\n"
+
+    return string
+
+
+def ValidateFriendlyNames() -> None:
+    '''Check the validity of each guid from the friendlynamedict in friendlynames.py'''
+    rowcounter = 0  # Used to track which row we are checking
+
+    for f in friendlynamedict:
```

Comment:
is friendlynamedict a global? I would perhaps use enumerate rather than a counter (https://docs.python.org/3/library/functions.html#enumerate)?

Reply:
Figured it out on the global thing.